src/Core/ServiceWrapper/Main.cs

```diff
@@ -276,6 +277,11 @@ protected override void OnStart(string[] args)
                 startarguments += " " + _descriptor.Arguments;
             }
 
+            // Converting newlines, line returns, tabs into a single 
+            // space. This allows users to provide multi-line arguments
+            // in the xml for readability.
+            startarguments = Regex.Replace(startarguments, @"[\n\r\t]+", " ");

```

Comment:
```suggestion
            startarguments = Regex.Replace(startarguments, @"\s+[\n\r\t]+\s+", " ");

```

Reply:
I'd guess it finalizes the proposal from @NextTurn in https://github.com/winsw/winsw/pull/479#issuecomment-608967443

Reply:
Indeed this is what I thought in https://github.com/winsw/winsw/pull/479#issuecomment-608967443. **This is acceptable.**

But now I'm thinking of a more conservative approach, which is to collapse only white spaces that are treated as delimiters by Windows.

Don't feel that you have to address it. This is what I planned for v3. In v2 you can use `<argument>`/`<startargument>`/`<stopargument>` without `s`.

(I know no one wants to put line breaks in command line arguments. But it's completed valid, unless we request users not to split one argument into lines.)

Reply:
@oleg-nenashev - the replacement pattern `@"\s+[\n\r\t]+\s+"` will only work if you have padding before and after the newline|linereturn|tab character.  So if you just have your argument followed by a newline, which I think will be a common case, that pattern will not match and remove the newline.

Reply:
Should be @"\s*[\n\r\t]+\s*".

Reply:
Ah, that makes a lot more sense.

Reply:
Pull request updated.